src/hotspot/share/nmt/vmatree.hpp

```diff
@@ -193,8 +246,35 @@ class VMATree {
 #endif
   };
 
+  struct RequestInfo {
+    position A, B;
+    StateType op;
+    MemTag tag;
+    SIndex callstack;
+    bool use_tag_inplace;
+    int op_to_index() const {
+      return
+            op == StateType::Released ? 0 :
+            op == StateType::Reserved && !use_tag_inplace ? 1 :
+            op == StateType::Committed ? 2 :
+            op == StateType::Reserved && use_tag_inplace ? 3 : -1;
+    }
+  };
+
  private:
   SummaryDiff register_mapping(position A, position B, StateType state, const RegionData& metadata, bool use_tag_inplace = false);
+  StateType get_new_state(const StateType existinting_state, const RequestInfo& req) const;
+  MemTag get_new_tag(const MemTag existinting_tag, const RequestInfo& req) const;
+  SIndex get_new_reserve_callstack(const SIndex existinting_stack, const StateType ex, const RequestInfo& req) const;
+  SIndex get_new_commit_callstack(const SIndex existinting_stack, const StateType ex, const RequestInfo& req) const;
+  void compute_summary_diff(const SingleDiff::delta region_size, const MemTag t1, const StateType& ex, const RequestInfo& req, const MemTag new_tag, SummaryDiff& diff) const;
+  void update_region(TreapNode* n1, TreapNode* n2, const RequestInfo& req, SummaryDiff& diff);
+  int state_to_index(const StateType st) const {
```

Comment:
Can't this just use the enum values? `return static_cast<int>(static_cast<uint8_t>(st));`

Reply:
Unfortunately not. Later in the main PR for porting VMATree to NMT, we needed to redefine `Commit` to be `3` and now it is `2`.